### PR TITLE
Adding Github command.

### PR DESCRIPTION
**Pull request by Jens-Co**
image

### CODE REVIEW

**Comment by onebeastchris** [comment]
yoooo, this is - again - sick! gone are the times when you had to manually search GitHub for a link

just to be sure, this just searches GitHub for repos matching the name? Or just the geyser/floodgate related one's?

**Comment by Camotoy** [comment]
I feel like this should have two slash command string options: mandatory string for repository name, and optional string for owner/organization defaulting to GeyserMC.

**Comment by Jens-Co** [comment]
> I feel like this should have two slash command string options: mandatory string for repository name, and optional string for owner/organization defaulting to GeyserMC.

https://github.com/GeyserMC/GeyserDiscordBot/commit/2d5b1f3606cfdba117edb96198f5ec55fe2703dc added an owner optional! Thanks.

File: src/main/java/org/geysermc/discordbot/util/BotHelpers.java [hunk 1]
```diff
@@ -329,4 +333,39 @@ public static String trim(String input, int length) {
...
+    /**
+     * Get a Github repository
+     *
+     * @param repository Input string to get repository
+     * @return Repository A repository if one was found, or null otherwise.
+     */
+    public static GHRepository getRepo(String repository, String owner) throws IOException {
+        GHRepository repo;
+        GHRepository defaultRepo = GeyserBot.getGithub().getRepository("GeyserMC/Geyser");
+        // if optional is used we get repo from optional first.
```
Comment (rtm516) [inline src/main/java/org/geysermc/discordbot/util/BotHelpers.java:346]:
```suggestion

        // if optional is used we get repo from optional first.
```

File: src/main/java/org/geysermc/discordbot/util/BotHelpers.java [hunk 2]
```diff
@@ -329,4 +333,39 @@ public static String trim(String input, int length) {
...
+        if (!owner.isEmpty()) {
+            try {
+                repo = GeyserBot.getGithub().getRepository(owner + "/" + repository);
+                return repo;
+            } catch (IOException e) {
+                // default to geyser repo.
+                return defaultRepo;
+            }
+        }
+        // if no owner is given search github repo's.
```
Comment (rtm516) [inline src/main/java/org/geysermc/discordbot/util/BotHelpers.java:356]:
```suggestion

        // if no owner is given search github repo's.
```

File: src/main/java/org/geysermc/discordbot/util/BotHelpers.java [hunk 3]
```diff
@@ -329,4 +333,39 @@ public static String trim(String input, int length) {
...
+                PagedSearchIterable<GHRepository> results = GeyserBot.getGithub().searchRepositories().q(matcherRepo.group(3)).list();
+                repo = results.toArray()[0];
+            } else {
+                repo = GeyserBot.getGithub().getRepository(matcherRepo.group(2) + matcherRepo.group(3));
+            }
+        } else {
+            // default to Geyser repo.
+            repo = defaultRepo;
+        }
+        return repo;
```
Comment (rtm516) [inline src/main/java/org/geysermc/discordbot/util/BotHelpers.java:369]:
```suggestion

        return repo;
```

**Review by rtm516 — CHANGES_REQUESTED** [review]
Just allowing the code to be a bit easier to read

**Comment by Jens-Co** [comment]
> Just allowing the code to be a bit easier to read

Makes sense thanks!

**Review by Tim203 — APPROVED** [review]

**Pull request merged**